File: models/store/IssuedItem.js
const { DataTypes } = require('sequelize');
const sequelize = require('../../config/database');

const IssuedItem = sequelize.define('IssuedItem', {
    id: {
        type: DataTypes.UUID,
        primaryKey: true,
        defaultValue: DataTypes.UUIDV4
    },
    item_id: {
        type: DataTypes.UUID,
        allowNull: false
    },
    batch_id: {
        type: DataTypes.UUID,
        allowNull: true
    },
    institution_id: {
        type: DataTypes.UUID,
        allowNull: true
    },
    issue_number: {
        type: DataTypes.STRING(50),
        unique: true,
        allowNull: false
    },
    quantity: {
        type: DataTypes.INTEGER,
        allowNull: false
    },
    issued_to: {
        type: DataTypes.UUID, // Staff ID
        allowNull: true
    },
    department_id: {
        type: DataTypes.UUID,
        allowNull: true
    },
    issued_by: {
        type: DataTypes.UUID, // Store staff ID
        allowNull: true
    },
    issue_date: {
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW
    },
    purpose: {
        type: DataTypes.TEXT,
        allowNull: true
    },
    status: {
        type: DataTypes.ENUM('issued', 'returned', 'partially_returned'),
        defaultValue: 'issued'
    },
    return_date: {
        type: DataTypes.DATE,
        allowNull: true
    },
    notes: {
        type: DataTypes.TEXT,
        allowNull: true
    }
}, {
    tableName: 'issued_items',
    timestamps: true,
});

// Define associations
IssuedItem.associate = (models) => {
    IssuedItem.belongsTo(models.Item, { foreignKey: 'item_id', as: 'item' });
    IssuedItem.belongsTo(models.ItemBatch, { foreignKey: 'batch_id', as: 'batch' });
};

module.exports = IssuedItem;

File: store-service/app/models/issued_item.py
"""IssuedItem model."""

import uuid
from datetime import datetime

from sqlalchemy import Column, DateTime, Enum, Integer, String, Text
from sqlalchemy.dialects.postgresql import UUID

from app.database import Base


class IssuedItem(Base):
    __tablename__ = "issued_items"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    item_id = Column(UUID(as_uuid=True), nullable=False)
    batch_id = Column(UUID(as_uuid=True), nullable=True)
    institution_id = Column(UUID(as_uuid=True), nullable=True)
    issue_number = Column(String(50), unique=True, nullable=False)
    quantity = Column(Integer, nullable=False)
    issued_to = Column(UUID(as_uuid=True), nullable=True)
    department_id = Column(UUID(as_uuid=True), nullable=True)
    issued_by = Column(UUID(as_uuid=True), nullable=True)
    issue_date = Column(DateTime(timezone=True), default=datetime.utcnow)
    purpose = Column(Text, nullable=True)
    status = Column(Enum("issued", "returned", "partially_returned", name="issue_status_enum"), default="issued")
    return_date = Column(DateTime(timezone=True), nullable=True)
    notes = Column(Text, nullable=True)

    created_at = Column("createdAt", DateTime(timezone=True), default=datetime.utcnow)
    updated_at = Column("updatedAt", DateTime(timezone=True), default=datetime.utcnow, onupdate=datetime.utcnow)
